d3/js/AliTV.js

```diff
@@ -959,6 +952,217 @@ AliTV.prototype.getOuterRadius = function() {
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterVisibleChromosomes = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	$.each(visibleChromosomes, function(key, value) {
+		if (that.filters.karyo.chromosomes[key].visible === true) {
+			filteredChromosomes[key] = value;
+		}
+	});
+	return filteredChromosomes;
+};
+
+/**
+ *This method should filter all chromosome which have no linkage information
+ * @param visibleChromosomes: the method gets all current visible chromosomes.
+ * @returns filteredChromosomes: the method returns only chromosomes which have linkage information
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterChromosomeWithoutLinkageInformation = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	$.each(visibleChromosomes, function(key, value) {
+		var currentChromosome = key;
+		var valueOfCurrentChromosome = value;
+		$.each(that.data.links, function(key, value) {
+			if (that.data.features[value.source].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false || that.data.features[value.target].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false) {
+				filteredChromosomes[currentChromosome] = valueOfCurrentChromosome;
+			}
+		});
+	});
+	return filteredChromosomes;
+};
+
+/**
+ * This method should filter all chromosome which have no visible links with the current configurations
+ * @param visibleChromosomes: the method gets all current visible chromosomes.
+ * @returns filteredChromosomes: the method returns only chromosomes which have visible links
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterChromosomeWithoutVisibleLinks = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	var allLinks = that.data.links;
+	var filteredLinks = that.filterLinksByIdentity(allLinks);
+	filteredLinks = that.filterLinksByLength(filteredLinks);
+	filteredLinks = that.filterLinksByAdjacency(filteredLinks);
+	$.each(visibleChromosomes, function(key, value) {
+		var currentChromosome = key;
+		var valueOfCurrentChromosome = value;
+		$.each(filteredLinks, function(key, value) {
+			if (that.data.features[value.source].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false || that.data.features[value.target].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false) {
+				filteredChromosomes[currentChromosome] = valueOfCurrentChromosome;
+			}
+		});
+	});
+	return filteredChromosomes;
+};
+
+/**
+ * This method is supposed to filter the order of chromosomes according to all visible chromosomes.
+ * @param visibleChromosomes: gets all visible chromosomes
+ * @return chromosomeOrder: returns the order of the visible chromosomes
+ * @author Sonja Hohlfeld
+ */
+AliTV.prototype.filterChromosomeOrder = function(visibleChromosomes) {
```

Comment:
I do not fully understand this function. Shouldn't the filters.chromosomeOrder array be used to determine the order of the chromosomes?